packages/bridge-ui-v2/src/libs/chain/chains.ts

```diff
@@ -135,42 +141,62 @@ export const chainUrlMap: Record<
 export const chainIcons = {
   [PUBLIC_L1_CHAIN_ID]: '/ethereum-chain.png',
   [PUBLIC_L2_CHAIN_ID]: '/taiko-chain.png',
-  [PUBLIC_L3_CHAIN_ID]: '',
+  [PUBLIC_L3_CHAIN_ID]: '/eldfell.svg',
 };
 
-export const chainContractsMap: Record<
-  string,
-  {
-    bridgeAddress: Address;
-    tokenVaultAddress: Address;
-    erc721VaultAddress: Address;
-    erc1155VaultAddress: Address;
-    crossChainSyncAddress: Address;
-    signalServiceAddress: Address;
-  }
-> = {
+type AddressConfig = {
+  bridgeAddress: Address;
+  erc20VaultAddress: Address;
+  erc721VaultAddress: Address;
+  erc1155VaultAddress: Address;
+  crossChainSyncAddress: Address;
+  signalServiceAddress: Address;
+};
+
+type ConfiguredChain = (typeof chains)[number]['id'];
+type RoutingConfig = Record<ConfiguredChain, AddressConfig>;
+
+export const routingContractsMap: Record<ConfiguredChain, RoutingConfig> = {
```

Comment:
As we also want to have L2<->L2 in the future or multiple L3s. We need to be a lot more flexbile. Maybe we could parse an array of objects instead? For example:

```
L2_CHAINS=[ 
 {
     chainId: 123
      bridgeAddress: address,
      erc20VaultAddress: address,
      erc721VaultAddress: addresss,
      erc1155VaultAddress: address,
      crossChainSyncAddress: address
      signalServiceAddress: address,
    } ,
    {
     chainId: 456
      bridgeAddress: address,
      erc20VaultAddress: address,
      erc721VaultAddress: addresss,
      erc1155VaultAddress: address,
      crossChainSyncAddress: address
      signalServiceAddress: address,
    } ,
    [...]
]
```

and a "link map" 
```
LINKED_CHAINS=[
chain123 : chain456,
chain:123 : chain789
]
```
But in a .env it's all just strings... Not ideal for complex structures like this.
We could use a config json file and just configure the path/name to that config in our .env

**But**: The above example also only works if we have singular signal services. The current deployment has separate ones for L1<->L2 and L2<->L3, and it could be the case for some other deployments in the future? Lot's to think about.

Any suggestions or comments @cyberhorsey @xiaodino @jscriptcoder ?

Reply:
I think using ```.env``` is not ideal. It's not easy to maintain

Reply:
Updated the example to the current state of my proposal.